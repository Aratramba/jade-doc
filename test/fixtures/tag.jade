//- @jadedoc
  name: some-tag

div.some-tag
  | this is some tag

div.some-other-tag
  | this is some other tag