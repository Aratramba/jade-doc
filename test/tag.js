'use strict';
/* global require */

var test = require('tape');
var jadeDoc = require('../');

test('tag', function(assert){
  assert.plan(4);

  var stream = jadeDoc({
    input: ['./test/fixtures/tag.jade']
  });

  stream.on('data', function(data){
    data = JSON.parse(data);

    var actual = data.meta.name;
    var expected = 'some-tag';
    assert.equal(actual, expected, 'Name should match name inside YAML data');

    actual = data.file;
    expected = 'test/fixtures/tag.jade';
    assert.equal(actual, expected, 'Filename should match the jade file');
    
    actual = data.source;
    expected = 'div.some-tag\n  | this is some tag';
    assert.equal(actual, expected, 'Source code block should be correct');

    actual = data.output;
    expected = '<div class=\"some-tag\">this is some tag</div>';
    assert.equal(actual, expected, 'Its html output should be correct');
  });
});