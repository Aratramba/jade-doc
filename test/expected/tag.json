{
  "some-tag": {
    "jade": {
      "comment": "  name: some-tag",
      "type": "tag",
      "indent": 0,
      "root": "div.some-tag",
      "code": "div.some-tag\n  | this is some tag",
      "file": "./test/fixtures/tag.jade"
    },
    "name": "some-tag",
    "html": "\n<div class=\"some-tag\">this is some tag</div>"
  }
}